### DISCUSSION TITLE
Support albums existing as one single FLAC file with Cue metadata embeded or in separate external file

### DISCUSSION THREAD

**Discussion by jpbaril** [music-assistant/support#2811] [Feature requests and ideas]
As already mentioned by someone else at https://github.com/music-assistant/hass-music-assistant/issues/2284, single file full albums in FLAC + Cue format are detected as a single track with duration of the full album instead of being split in "virtual" separate tracks.

This works in most of music players I ever used in the last 20 years and apparently it even works in LMS which I understand is the basis of MASS.

**Comment by OzGav** [comment]
MA is not based on LMS it has been written from scratch. Request is duly noted, will see how many votes it gets!

**Reply by awaldram** [reply]
Just started playing with MA , what a superb work .. well done. love the effort on quality. Busy deciding what to build as a battery power player probably going some variant on esp32squeezelite, Was very surprised by lack of cue file parsing given that .cue files are just as important as .m3u and potentially more so being cross platform (mp3/FLC/Ogg) whereas m3u is really only mp3 (by design) though guess they have different purposes.

All my HD and many CD rips are single file as this decision was take as a 'quality' decision allowing 100% accurate bit copy of the CD/DVD.

On the playlist front I have have a major issue (no laughing) , Never having utilised them just let them be in my 'ripped CD' folder from my original rip. But now MA has had its way I now have hundreds of playlists called 'playlist' 
:1st_place_medal:  on the plus side they all work :).

Mainly responded to thank you for a really great project. may rip a subset lower quality MP3 in separate tracks  to get round the lack of .cue  support as imagine its very low on the todo list if it even made it.

**Reply by tedl57** [reply]
Cuesheet support would be a killer extension!  

i just installed MA in HA and love it so far!  i love that MA integrates so well with all the existing HA functionality (media players, media, etc.)  

i currently have 2300+ ripped cd's with embedded cuesheets with a total of 30k+ tracks (that's a big jukebox :-)

i've cobbled together an integration with (very dated but nice) Clementine Music Player because it supports cuesheets and has an api for building playlists, etc. from all the tracks.  i'd drop this effort in a heartbeat if MA supported cuesheets!  Thanks!

**Comment by arigit** [comment]
I'd love to give music-assistant a shot (I have been using HA at home for 5-6 years now), but this missing feature is a blocker for me. My entire music collection is packaged as single-file FLAC + external cuesheet file. This works really well for Classical in particular and makes it really easy in my case to keep tagging consistency. I understand this is kind of niche but there are many audiophiles out there using it.

For the time being I use a few headless raspberry pi + high quality DACs or digital SPDIF cards  (hifiberry), in the latter case  to output bit-perfect 24bit/96Khz or higher to amp-integrated speaker sets. 

The RPIs run a headless Kodi (OSMC) install  with a centralized database and NFS share for the audio media. I would very much like to replace the SW setup with a simpler music-focused client running on the RPi's but so far, I found no better open source option, Kodi processes my thousands of album/cuesheets like a champ, it's really solid, I've been using this setup (with some hardware variations) for more than 10 years now. The kodi music dev has been so responsive and Kodi's code is so solid. All other solutions out there fail in different ways, typically with poor or no proper support of external cuesheets.

Looking forward to Music-assistant including this feature, supporting external DACs & HifiBerry DIGIs, and becoming the go-to option for digital audio streaming endpoints.

**Comment by zhaoxitao** [comment]
I've been using MA for a while now, and I have to say it's fantastic. I'm an audiophile, and to make my CDs more collectible (to prevent them from being damaged prematurely), I've saved all of my 1000+ CDs as digital files in APE+CUE format. Most players support CUE files well, and if MA could support it too, it would be excellent.

**Comment by nico1080** [comment]
I confirm all comment, I just installed music assistant and I am really impressed by how simple/powerful it is.
However my music library is 90% under cue format (Flac, APE etc) so it is barely usable without cue recognition

**Comment by OzGav** [comment]
Support for external cue sheets will be available in 2.10.0b3

**Reply by OzGav** [reply]
2.10.0b3 has been released

**Comment by tedl57** [comment]
Excellent, thank you!

How do you think MA will perform with 30k tracks in 2400 cue sheets?

**Reply by OzGav** [reply]
I didn’t not test that scenario but it should complete fine in time. Before getting too far in I recommend looking at the docs to see the info on cue sheet contents for MA and do a test run to make sure everything works ok. Having a minimal cue sheet is the same as minimal tags and we recommend against that.

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
